# Business & Activity - Supporting local businesses surrounding Tourist attractions Usecase

## Authored by:  Kruthi Shetty and Vinit Shetty

**Duration:** 120 mins <br>
**Level:** Intermediate <br>
**Pre-requisite Skills:** Python, Data Engineering and Analysis

### Scenario

**As a tourist I want to make the most of the time in the city of Melbourne and have a more enjoyable and fulfilling experience.** <br>
When visiting Melbourne, tourists are often eager to uncover new and exciting places to visit, in order to add depth and interest to their itinerary. By researching lesser-known attractions or asking locals for recommendations, travelers can discover hidden gems that may not have been on their radar initially. Additionally, it's important to plan one's route effectively in order to avoid wasting time and money on unnecessary transportation. By mapping out a logical itinerary, visitors can ensure that they are able to see as much of the city as possible within their allotted time frame, without missing out on any must-see sights.

**As a cafe, restaurant, or bistro owner in Melbourne, I understand the value of catering to tourists who are looking to make the most of their time in the city. By recognizing and responding to the needs and interests of visitors, I can build a loyal customer base and create a sustainable business model that benefits both myself and the local tourism industry.**<br> 
The primary objective of business owners is to enhance customer engagement and drive revenue growth by delivering an exceptional and unforgettable experience to visitors. This can result in a surge of foot traffic and revenue generation for the cafe, restaurant, and bistro owners. In the end, the aim of business owners is to develop a feasible business model that not only serves their best interests but also provides substantial benefits to their customers and the neighboring community.

**What Will This Use Case Teach You?**

At the end of this use case you will:

- Learn how to Load the dataset into a Pandas dataframe.
- Clean the data by removing any unnecessary columns and handling missing values
- Use the data to create heatmaps to show the density of landmarks or places of interest in  different areas of Melbourne
- Add valuable information about each tourist attractions or place of interest to the visualizations, such as opening hours, ticket prices, and ratings and reviews from previous visitors.

#### Introduction

Melbourne is known for its vibrant arts and culture scene, with numerous museums, galleries, and theaters. The city is also famous for its food and wine culture, with a diverse range of restaurants, cafes, and bars.It has a thriving coffee culture and is known for its brunch spots. Overall, Melbourne offers a unique combination of culture, food, nature, and sports, making it a popular destination for tourists from around the world.

The objective of this use case is to offer tourists a comprehensive overview of the popular tourist spots and interesting locations in Melbourne, coupled with valuable insights to help them organize their trip effectively. By displaying the attractions on a map, visitors can effortlessly determine their location and proximity to one another, which makes planning their itinerary and optimizing their time in Melbourne a more convenient task. Through the information and guidance provided for each attraction, tourists can make informed decisions regarding which places to visit, which helps them avoid potential disappointments.

Moreover, this use case also benefits the nearby cafes, restaurants, and bistros owners as they will be featured on the map along with the landmarks, leading to an increase in revenue and customer footfall for their establishments.

#### Datasets List

- Landmarks and places of interest, including schools, theatres, health services, sports facilities, places of worship, galleries and museums.
- Public artworks, fountains and monuments.
- Public memorials and sculptures.
- Cafe, restaurant and bistro seats.

### Roadmap

#### Contents
1. Import Libraries
2. Connect and Test Datasets
3. Analysis of Datasets
    - 3.1. Landmarks and places of interest
    - 3.2. Public artworks, fountains and monuments
    - 3.3. Public memorials and sculptures
    - 3.4. Cafe, restaurant and bistro seats
4. Combining all Visualisations and Data
5. Findings and Opportunities
6. Thank You!

### 1. Import important Libraries

In [21]:
# Standard
import os
import json

# Data import
import urllib
from urllib.request import urlopen 
import requests
from sodapy import Socrata

# Data manipulation
import pandas as pd

# Plotting 
import plotly.graph_objs as go
import plotly.express as px

### 2. Connect and Test datasets

In [9]:
# Landmarks and places of interest data
lan_url = 'https://data.melbourne.vic.gov.au/api/v2/catalog/datasets/landmarks-and-places-of-interest-including-schools-theatres-health-services-spor/exports/json?limit=-1&offset=0&timezone=UTC'
r = requests.get(lan_url)
response = r.json()
lan_data = pd.DataFrame(response)

# Public artworks, fountains and monuments data
Pub_url = 'https://data.melbourne.vic.gov.au/api/v2/catalog/datasets/public-artworks-fountains-and-monuments/exports/json?limit=-1&offset=0&timezone=UTC'
r = requests.get(Pub_url)
response = r.json()
Pub_data = pd.DataFrame(response)

# Public memorials and sculptures data
mem_url = 'https://data.melbourne.vic.gov.au/api/v2/catalog/datasets/public-memorials-and-sculptures/exports/json?limit=-1&offset=0&timezone=UTC'
r = requests.get(mem_url)
response = r.json()
mem_data = pd.DataFrame(response)

# Cafe, restaurant and bistro seats data
cafe_url = 'https://data.melbourne.vic.gov.au/api/v2/catalog/datasets/cafes-and-restaurants-with-seating-capacity/exports/json?limit=-1&offset=0&timezone=UTC'
r = requests.get(cafe_url)
response = r.json()
cafe_data = pd.DataFrame(response)

Next, we will look at one specific dataset to better understand its structure and how we can use it. For this exercise, we will observe the Landmarks and places of interest dataset, specifically, it's first ten rows.

In [19]:
# Print details of data
print(f'The shape of the dataset is:{lan_data.shape}')
print()
print('The first ten rows of this dataset are:')

# Print the first 10 rows of data
lan_data.head(10)

The shape of the dataset is:(242, 4)

The first ten rows of this dataset are:


theme           sub_theme                     feature_name  \
0  Place of Worship              Church                St Francis Church   
1  Place of Worship              Church                  St James Church   
2  Place of Worship              Church        St Mary's Anglican Church   
3  Place of Worship              Church                     Scots Church   
4  Place of Worship              Church      St Michael's Uniting Church   
5  Place of Worship              Church            Greek Orthodox Church   
6  Place of Worship              Church          North Melbourne Uniting   
7  Place of Worship              Church  South Yarra Presbyterian Church   
8  Place of Worship           Synagogue         East Melbourne Synagogue   
9         Transport  Transport Terminal                Port of Melbourne   

                                        co_ordinates  
0  {'lon': 144.962422614541, 'lat': -37.811884783...  
1  {'lon': 144.952468571683, 'lat': -37.810128120...  
2  {'lon': 144.953761537074, 'lat': -37.803166367...  
3  {'lon': 144.96855105335, 'lat': -37.8145687802...  
4  {'lon': 144.969174036096, 'lat': -37.814385132...  
5  {'lon': 144.978259089269, 'lat': -37.808806466...  
6  {'lon': 144.947671538375, 'lat': -37.803553847...  
7  {'lon': 144.98562699348, 'lat': -37.8407473645...  
8  {'lon': 144.97422190954, 'lat': -37.809113728917}  
9  {'lon': 144.91753432375, 'lat': -37.8137384362...

We can see that there are 242 records and 4 fields describing each record. 

Each record can be broken down into the following fields:

**theme:** Main Theme of the landmark. e.g. Places of Interest.

**sub_theme:** Exact Identification of the landmark. e.g Church or Synagogue.

**feature_name:** Name of the landmark. e.g. St Francis Church, East Melbourne Synagogue.

**co_ordinates:** Data regarding the geographical location of a place, determined by its latitude and longitude. 

**Awesome! After taking a look at one of the Melbourne datasets we can see the overall structure and contents. Lets now begin our analysis of each individual dataset.

### 3. Analysis of the dataset

We are now going analyse each individual dataset so we can generate useful information and visualisations about them. This will assist us when producing final interactive maps and predictions later on.

#### 3.1 Landmarks and Places of Interest Dataset

In [24]:
lan_data

theme                                 sub_theme  \
0    Place of Worship                                    Church   
1    Place of Worship                                    Church   
2    Place of Worship                                    Church   
3    Place of Worship                                    Church   
4    Place of Worship                                    Church   
..                ...                                       ...   
237  Education Centre  School - Primary and Secondary Education   
238            Retail                          Department Store   
239            Retail                          Department Store   
240   Health Services                          Medical Services   
241         Mixed Use                     Retail/Office/Carpark   

                       feature_name  \
0                 St Francis Church   
1                   St James Church   
2         St Mary's Anglican Church   
3                      Scots Church   
4       St Michael's Uniting Church   
..                              ...   
237  Melbourne Girls Grammar School   
238                            Myer   
239                     David Jones   
240          Mercy Private Hospital   
241               ANZ 'Gothic' Bank   

                                          co_ordinates  
0    {'lon': 144.962422614541, 'lat': -37.811884783...  
1    {'lon': 144.952468571683, 'lat': -37.810128120...  
2    {'lon': 144.953761537074, 'lat': -37.803166367...  
3    {'lon': 144.96855105335, 'lat': -37.8145687802...  
4    {'lon': 144.969174036096, 'lat': -37.814385132...  
..                                                 ...  
237  {'lon': 144.985089428348, 'lat': -37.831536451...  
238  {'lon': 144.963855087868, 'lat': -37.813591198...  
239  {'lon': 144.964373486798, 'lat': -37.813312726...  
240  {'lon': 144.984435746587, 'lat': -37.811896809...  
241  {'lon': 144.961673719242, 'lat': -37.816158066...  

[242 rows x 4 columns]

Map - using co ordinates - get lat and long - Using Folium - Once we have the map - Name of the landmark - sub_theme as Icon - Theme as a whole cluster 

#### 3.2 Public Artworks, Fountains and Monuments Dataset

In [26]:
Pub_data.head(2)

asset_type       name                          xorg  \
0        Art   Blowhole  Beveridge Williams Surveyors   
1        Art  Bandstand             City of Melbourne   

                                xsource                         address_point  \
0                          Field Survey       13 Harbour Esplanade, DOCKLANDS   
1  MCC - Ortho Image March 2005 - Final  200 Little Collins Street, MELBOURNE   

           artist alternate_name art_date mel_way_ref  respective_author  \
0  Duncan Stemler           None     2005       2E_G8  City of Melbourne   
1     George Dodd           None     1864       2F_G3  City Of Melbourne   

                                 structure  \
0  Stainless-steel and aluminium sculpture   
1                       Concrete bandstand   

                                        co_ordinates     easting     northing  
0  {'lon': 144.946871022845, 'lat': -37.822018216...  319296.292  5811946.283  
1  {'lon': 144.96733868317, 'lat': -37.8140548689...  321078.689  5812869.353

Need - asset_type(Icon) - Name , artist , art_date , structure (description) , co_ordinates ( To Map) 
do not need - xorg,	xsource,address_point , alternate_name , mel_way_ref , respective_author , easting	northing

#### 3.3 Public Memorials and Sculptures Dataset

#### 3.4 Cafe, Restaurant and Bistro Seats Dataset